coherent_rates/isf.py

```diff
@@ -185,11 +189,110 @@ def get_boltzmann_isf(
     }
+
+
+def get_boltzmann_isf(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    direction: tuple[int] = (1,),
+    *,
+    n_repeats: int = 10,
+) -> ValueList[_AX0Inv]:
+    hamiltonian = get_hamiltonian(system, config)
+    return get_boltzmann_isf_from_hamiltonian(
+        hamiltonian,
+        config.temperature,
+        times,
+        direction,
+        n_repeats=n_repeats,
+    )
+
+
+def get_boltzmann_isf_stats(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    direction: tuple[int] = (1,),
+    *,
+    average_over: int = 10,
+) -> StatisticalValueList[_AX0Inv]:
+    isf_data = np.zeros((average_over, times.n), dtype=np.complex128)
+    hamiltonian = get_hamiltonian(system, config)
+    operator = get_periodic_x_operator(hamiltonian["basis"][0], direction)
+
+    for _i in range(average_over):
+        state = _get_random_boltzmann_state_from_hamiltonian(
+            hamiltonian,
+            config.temperature,
+        )
+        data = _get_isf_from_hamiltonian(hamiltonian, operator, state, times)
+        isf_data[_i, :] = data["data"]
+    mean = np.mean(isf_data, axis=0, dtype=np.complex128)
+    sd = np.std(isf_data, axis=0, dtype=np.complex128)
+    return {
+        "data": mean,
+        "basis": times,
+        "standard_deviation": sd,
+    }
+
+
+def truncate_value_list(
+    values: ValueList[EvenlySpacedTimeBasis[int, int, int]],
+    index: int,
+) -> ValueList[EvenlySpacedTimeBasis[int, int, int]]:
+    data = values["data"][0 : index + 1]
+    new_times = EvenlySpacedTimeBasis(index + 1, 1, 0, values["basis"].times[index])
+    return {
+        "basis": new_times,
+        "data": data,
+    }
+
+
+def get_ak_data_1d(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    *,
+    n_points: int = 8,
+) -> tuple[
+    np.ndarray[tuple[int], np.dtype[np.float64]],
+    np.ndarray[tuple[int], np.dtype[np.float64]],
+]:
+    length = get_potential(system)["basis"].delta_x_stacked[0] * config.shape[0]
+    kpoints = config.shape[0] * config.resolution[0]
+    xdata = np.floor(np.linspace(int(kpoints / 10), int(kpoints * 0.45), n_points))
+    ydata = np.zeros(xdata.shape)
+    hamiltonian = get_hamiltonian(system, config)
+    for i in range(n_points):
+        isf = get_boltzmann_isf_from_hamiltonian(
+            hamiltonian,
+            config.temperature,
+            times,
+            (xdata[i],),
+            n_repeats=20,
+        )
+        idx = times.n - 1
+        data = np.abs(isf["data"])
+        for _i in range(times.n - 1):
+            if data[_i + 1] > data[_i]:
+                idx = _i
+                break
+        truncated_isf = truncate_value_list(isf, idx)
+        ydata[i] = 1 / get_gaussian_fit(truncated_isf)[2]
+        times = EvenlySpacedTimeBasis(
+            times.n,
+            times.step,
+            times.offset,
+            times.times[idx] * 1.2,
```

Comment:
does this only work if your delta k spacing is small - the 1.2 factor should maybe depend on this??